ha/core/action_handler/action_handler.py

```diff
@@ -140,20 +138,34 @@ def act(self, event: HealthEvent, action: list) -> None:
             None
         """
         Log.info(f"Default action handler with Event: {event} and actions : {action}")
-        if HEALTH_MON_ACTIONS.PUBLISH_ACT.value in action and len(action) == 1:
+        if len(action) == 0:
+            return
+        elif HEALTH_MON_ACTIONS.PUBLISH_ACT.value in action and len(action) == 1:
             self.publish_event(event)
         else:
-            raise InvalidAction()
+            raise InvalidAction(f"{action} is invalid, DefaultActionHandler will only allow {HEALTH_MON_ACTIONS.PUBLISH_ACT.value}")
 
-
-class NodeFailureActionHandler(ActionHandler):
+class NodeActionHandler(DefaultActionHandler):
```

Comment:
Why this class inherits DefaultActionHandler instead of ActionHandler now?

Reply:
yes it is ActionHandler it forgot to change after discussion with Archana
updated